packages/lib-classifier/src/components/Classifier/Classifier.spec.js

```diff
@@ -0,0 +1,431 @@
+    })
+
+    describe('task answers', function () {
+      it('should be displayed', function () {
+        expect(taskAnswers).to.have.lengthOf(workflow.tasks.T0.answers.length)
+      })
+
+      it('should be linked to the task', function () {
+        taskAnswers.forEach(radioButton => {
+          expect(radioButton.name).to.equal('T0')
+        })
+      })
+
+      it('should be enabled', function () {
+        taskAnswers.forEach(radioButton => {
+          expect(radioButton.disabled).to.be.false()
+        })
+      })
+    })
+  })
+
+  describe('with permission to view an inactive workflow', function () {
+    const allowedRoles = ['owner', 'collaborator', 'tester']
+    allowedRoles.forEach(function (role) {
+      describe(role, function () {
+        before(async function () {
+          const roles = [role]
+          const subjectSnapshot = SubjectFactory.build({ locations: [{ 'image/png': 'https://foo.bar/example.png' }] })
+          const workflowSnapshot = branchingWorkflow
+          const projectSnapshot = ProjectFactory.build({
+            links: {
+              active_workflows: [],
+              workflows: [workflowSnapshot.id]
+            }
+          })
+          sinon.stub(panoptes, 'get').callsFake((endpoint, query, headers) => {
+            switch (endpoint) {
+              case '/field_guides': {
+                const field_guides = []
+                return Promise.resolve({ body: { field_guides }})
+              }
+              case '/project_preferences': {
+                const project_preferences = []
+                return Promise.resolve({ body: { project_preferences }})
+              }
+              case '/project_roles': {
+                const project_roles = [{ roles }]
+                return Promise.resolve({ body: { project_roles }})
+              }
+              case '/subjects/queued': {
+                const subjects = [subjectSnapshot, ...Factory.buildList('subject', 9)]
+                return Promise.resolve({ body: { subjects }})
+              }
+            }
+          })
+          sinon.stub(panoptes, 'post').callsFake((...args) => {
+            return Promise.resolve({ headers: {}, body: { project_preferences: []}})
+          })
+          const checkBearerToken = sinon.stub().callsFake(() => Promise.resolve('mockAuth'))
+          const checkCurrent = sinon.stub().callsFake(() => Promise.resolve({ id: 123, login: 'mockUser' }))
+          const authClient = { ...defaultAuthClient, checkBearerToken, checkCurrent }
+          const client = { ...defaultClient, panoptes }
+          const store = RootStore.create({}, { authClient, client })
+          render(
+            <Classifier
+              classifierStore={store}
+              project={projectSnapshot}
+              workflowID={workflowSnapshot.id}
+              workflowSnapshot={workflowSnapshot}
+              workflowVersion={workflowSnapshot.version}
+            />,
```

Comment:
In this test suite and the following one, I noticed that Classifier is rendered without a `subjectID`. I think it would be valuable to illustrate somewhere in lib-classifier what we expect Classifier to do when a `subjectID` exists and when it doesn't. That would probably mean testing the side effects of `onURLChange()`. Is that outside the scope here and instead tested in WorkflowStore?

Reply:
That’s a good point. I don’t think we’ve got any live projects that use `subjectID`. When it’s present, subject selection should show you exactly that subject ie. you can select which subject to work on. 

In the dev classifier, it’s useful to debug specific subjects, when you know the ID.

Reply:
I've updated all the tests to remove the subject IDs, as they aren't used by the test workflow.